File: cpp/examples/create_template.cpp
// Example: Creating an APR template in C++
#include <promptresponse/api.hpp>
#include <promptresponse/serialization.hpp>
#include <promptresponse/validation.hpp>
#include <iostream>

using namespace promptresponse;

int main() {
    // Build a contact form template
    auto document = TemplateBuilder("Contact Form", "contact-v1")
        .setDescription("Simple contact information form")
        .setAuthor("Example Corp")
        .setVersion("1.0")
        .addSection("Personal Information")
            .addPrompt("Full Name", "text", "John Doe")
            .addPrompt("Email Address", "email", "john@example.com")
            .addPrompt("Phone Number", "phone", "+1-555-0100")
            .done()
        .addSection("Message", "Your message to us")
            .addPrompt("Subject", "text")
            .addPrompt("Message", "multiline")
            .done()
        .build();

    // Validate
    auto result = AprValidator::validate(document);
    if (result.isValid) {
        std::cout << "✓ Validation passed\n";
    } else {
        std::cout << "✗ Validation failed:\n";
        for (const auto& error : result.errors) {
            std::cout << "  - " << error.field << ": " << error.message << "\n";
        }
        return 1;
    }

    // Save to file
    std::string outputPath = "contact-form.aprt";
    AprSerializer::saveFile(document, outputPath);
    std::cout << "✓ Template saved to: " << outputPath << "\n";
    std::cout << "  Sections: " << document.sections.size() << "\n";
    std::cout << "  Total prompts: " << document.getAllPrompts().size() << "\n";

    return 0;
}

File: rust/examples/create_template.rs
//! Example: Creating an APR template in Rust

use promptresponse::{TemplateBuilder, validation, serialization};

fn main() -> Result<(), Box<dyn std::error::Error>> {
    // Build a contact form template
    let document = TemplateBuilder::new("Contact Form", "contact-v1")
        .description("Simple contact information form")
        .author("Example Corp")
        .version("1.0")
        .section("Personal Information")
            .prompt("Full Name", "text")
            .prompt("Email Address", "email")
            .prompt("Phone Number", "phone")
            .done()
        .section("Message")
            .prompt("Subject", "text")
            .prompt("Message", "multiline")
            .done()
        .build();

    // Validate
    let result = validation::validate(&document);
    if result.is_valid {
        println!("✓ Validation passed");
    } else {
        println!("✗ Validation failed:");
        for error in &result.errors {
            println!("  - {}: {}", error.field, error.message);
        }
        return Ok(());
    }

    // Save to file
    let output_path = "contact-form.aprt";
    serialization::save_file(&document, output_path)?;
    println!("✓ Template saved to: {}", output_path);
    println!("  Sections: {}", document.sections.len());
    println!("  Total prompts: {}", document.get_all_prompts().len());

    Ok(())
}
